frontend/src/v5/ui/routes/dashboard/projects/containers/uploadFileForm/uploadFileForm.component.tsx

```diff
@@ -71,12 +68,14 @@ export const UploadFileForm = ({ openState, onClickClose }: IUploadFileForm): JS
 	});
 
 	useEffect(() => {
-		if (!isUploading) reset();
-	}, [isUploading]);
+		remove();
+		setSelectedIndex(null);
+		setIsUploading(false);
+	}, [open]);
 
 	useEffect(() => {
-		if (fields.some((field) => filesizeTooLarge(field.file))) {
-			setError('uploads', { type: 'custom', message: '' });
+		if (fields.some(({ file }) => filesizeTooLarge(file))) {
```

Comment:
I am not sure if it would make the code cleaner in the overall, but for this goal you could use a `useState` variable, like `filesTooLarge` and here you just call 
```ts
setFilesTooLarge(filesTooLarge(fields.some(({ file }) => filesizeTooLarge(file)))
```

and then use that variable to set the disableButton disabled state.

This way you don't need to alter the form errors. But again, I am not sure in the overall it would make the code much cleaner or just cleaner, so feel free to ignore this comment in case

Reply:
This let me clean up the code quite a bit, thanks!